### PR TITLE
Typespec EGv2 Clean CI

### PR DESCRIPTION
**Pull request by l0lawrence**

### CODE REVIEW

File: specification/eventgrid/Azure.Messaging.EventGrid/examples/2023-06-01-preview/PublishExample.json
Comment (l0lawrence) [inline specification/eventgrid/Azure.Messaging.EventGrid/examples/2023-06-01-preview/PublishExample.json:1]: have messaged service team about getting examples for operations -- made this one in the meantime

**Review by mikeharder — CHANGES_REQUESTED** [review]

File: specification/eventgrid/Azure.Messaging.EventGrid/main.tsp
Comment (johanste) [inline specification/eventgrid/Azure.Messaging.EventGrid/main.tsp:220]: There should be a Http.Body<T> that you can/should use.

Comment (l0lawrence) [reply specification/eventgrid/Azure.Messaging.EventGrid/main.tsp:220]: #24371

**Comment by mikeharder** [comment]
@l0lawrence: We are requiring specs to be formatted by `tsp format` as of https://github.com/Azure/azure-rest-api-specs/pull/24041.  Running `tsp format **/*.tsp` on your spec folder should be sufficient to comply.

**Review by mikeharder — APPROVED** [review]
Approved by me as long as CI is passing

File: specification/eventgrid/Azure.Messaging.EventGrid/main.tsp
Comment (johanste) [inline specification/eventgrid/Azure.Messaging.EventGrid/main.tsp:275]: Nitpick; specifying the charset for content-type application/json is a bit redundant since json is supposed to be in UTF-8. Does the service actually require us to include the charset? If so, we should open a bug...

File: specification/eventgrid/Azure.Messaging.EventGrid/main.tsp
Comment (johanste) [inline specification/eventgrid/Azure.Messaging.EventGrid/main.tsp:277]: Is there a separate PR for better documentation?

**Review by mikeharder — CHANGES_REQUESTED** [review]
We'd like to start enforcing `tsp format` in CI, but this spec is blocked until this issue is fixed:

https://github.com/microsoft/typespec/issues/2060

If you run `tsp format` on the files in this spec in main, it will move comments where they don't belong.

**Review by mikeharder — APPROVED** [review]
LGTM as long as it's passing CI.  You may want another reviewer to sign-off on the content changes.

File: specification/eventgrid/Azure.Messaging.EventGrid/main.tsp
Comment (l0lawrence) [reply specification/eventgrid/Azure.Messaging.EventGrid/main.tsp:275]: Tested without and worked, will remove

File: specification/eventgrid/Azure.Messaging.EventGrid/main.tsp
Comment (l0lawrence) [reply specification/eventgrid/Azure.Messaging.EventGrid/main.tsp:277]: #24473

File: specification/eventgrid/Azure.Messaging.EventGrid/tspconfig.yaml
Comment (mikeharder) [inline specification/eventgrid/Azure.Messaging.EventGrid/tspconfig.yaml:24]:
```suggestion
    package-dir: "Azure.Messaging.EventGrid"
```

Should this align with the folder already in the net repo?

https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/eventgrid/Azure.Messaging.EventGrid

**Pull request merged**
